### DISCUSSION TITLE
WRE3 changed with ESP8266 12s, now what?

### DISCUSSION THREAD

**Discussion by iMatux** [arendst/Tasmota#14478] [Q&A] [answered]
Hello, first thing to say: Be patient with me, I'm not hardware expert, my field is IT. And sorry for my English, I'm Italian.
A year ago I tried to convert this Tuya product, it's called with different names like Moes Ufo-R2-RF or only RF-R2 with no brand name. 
However, once opened I found out a WRE3 chip which is a RTL and not an ESP. A few weeks ago I bought 2 ESP8266 12s (I knew that I will be brick one) which  have the same pinout as WRE3, so I flashed it with Tasmota using a CH2430G and soldered on pcb instead of original chip.
Tasmota is working great, now the main problem is set correctly GPIO on Module / Template, I really don't know how to read(?) pcb and which does what.
I think Nuvoton ms51fb9ae is the controller for RF, but I'm not sure. There is also a "Key 2" switch which I don't get what is it for.
If you want to help me I would really appreciated it! I'm here if you need any info about this product.
![IMG_2494](https://user-images.githubusercontent.com/23347512/149675694-b0da5b36-5660-46ad-9ab8-3061636315e9.JPG)

![IMG_2496](https://user-images.githubusercontent.com/23347512/149675683-d706e3e5-cc1a-476e-abd4-10f6919e01d8.JPG)

**Comment by barbudor** [comment, marked as answer]
Wow, 4 antennas + IR leds ?
RF and IT bridge
Should you have chose a ESP-C3-12F you would also have Bluetooth in the same 🤣

The Nuvoton is likely to be a TuyaMCU so you should probably start by entering a template like this one in Configuration => Others
`{"NAME":"Tuya Device","GPIO":[0,2272,0,2304,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":54}`
And check the boxe "Activate" then Save.
After reboot, go to the console and you should see messages about TuyaReceived

Still, it would be good that you try to reconsrtuct as much as possible the diagram of the device.
Try to find where each GPIO of the ESP goes.
You can do that with your eyes and, if you have amultimeter, those generally have a contact tester that can beep where there is continuity (often with a diode symbol like `->|-`). That can be helpfull to understand where is signal is going.

Once you have more details, it will be easier to help you.

**Comment by RomanZw** [comment]
Did this story have a happy ending? I have the same device.
